masonry/src/widget/zstack.rs

```diff
@@ -0,0 +1,324 @@
+            (VerticalAlignment::Top, HorizontalAlignment::Trailing) => Self::TopTrailing,
+            (VerticalAlignment::Center, HorizontalAlignment::Leading) => Self::Leading,
+            (VerticalAlignment::Center, HorizontalAlignment::Center) => Self::Center,
+            (VerticalAlignment::Center, HorizontalAlignment::Trailing) => Self::Trailing,
+            (VerticalAlignment::Bottom, HorizontalAlignment::Leading) => Self::BottomLeading,
+            (VerticalAlignment::Bottom, HorizontalAlignment::Center) => Self::Bottom,
+            (VerticalAlignment::Bottom, HorizontalAlignment::Trailing) => Self::BottomTrailing,
+        }
+    }
+
+    pub fn vertical(self) -> VerticalAlignment {
+        match self {
+            Alignment::Center | Alignment::Leading | Alignment::Trailing => {
+                VerticalAlignment::Center
+            }
+            Alignment::Top | Alignment::TopLeading | Alignment::TopTrailing => {
+                VerticalAlignment::Top
+            }
+            Alignment::Bottom | Alignment::BottomLeading | Alignment::BottomTrailing => {
+                VerticalAlignment::Bottom
+            }
+        }
+    }
+
+    pub fn horizontal(self) -> HorizontalAlignment {
+        match self {
+            Alignment::Center | Alignment::Top | Alignment::Bottom => HorizontalAlignment::Center,
+            Alignment::Leading | Alignment::TopLeading | Alignment::BottomLeading => {
+                HorizontalAlignment::Leading
+            }
+            Alignment::Trailing | Alignment::TopTrailing | Alignment::BottomTrailing => {
+                HorizontalAlignment::Trailing
+            }
+        }
+    }
+}
+
+impl Default for Alignment {
+    fn default() -> Self {
+        Self::Center
+    }
+}
+
+impl Default for VerticalAlignment {
+    fn default() -> Self {
+        Self::Center
+    }
+}
+
+impl Default for HorizontalAlignment {
+    fn default() -> Self {
+        Self::Center
+    }
+}
+
+impl From<Alignment> for VerticalAlignment {
+    fn from(value: Alignment) -> Self {
+        value.vertical()
+    }
+}
+
+impl From<Alignment> for HorizontalAlignment {
+    fn from(value: Alignment) -> Self {
+        value.horizontal()
+    }
+}
+
+impl From<(VerticalAlignment, HorizontalAlignment)> for Alignment {
+    fn from((vertical, horizontal): (VerticalAlignment, HorizontalAlignment)) -> Self {
+        Alignment::new(vertical, horizontal)
+    }
+}
+
+impl From<VerticalAlignment> for Alignment {
+    fn from(vertical: VerticalAlignment) -> Self {
+        Alignment::new(vertical, HorizontalAlignment::Center)
+    }
+}
+
+impl From<HorizontalAlignment> for Alignment {
+    fn from(horizontal: HorizontalAlignment) -> Self {
+        Alignment::new(VerticalAlignment::Center, horizontal)
+    }
+}
+
+// --- MARK: IMPL ZSTACK ---
+impl ZStack {
+    pub fn new() -> Self {
+        ZStack {
+            children: Vec::new(),
+            alignment: Alignment::default(),
+        }
```

Comment:
```suggestion
        Self::default()
```

Reply:
I even omit these kind of `new` implementations nowadays, as they're kind of redundant.

Reply:
I find them useful occasionally - Rust-analyzer makes the `new` implementation much more obvious than a `::default`

Reply:
Changed in later commit